cpp/tests/utilities/column_utilities.cu

```diff
@@ -47,7 +47,12 @@ void column_property_comparison(cudf::column_view const& lhs, cudf::column_view
 }
 
 void expect_column_properties_equal(column_view const& lhs, column_view const& rhs) {
+  auto lhs_col = cudf::column(lhs);
```

Comment:
What's going on here?

Reply:
This PR fixes an issue with materializing a string column. I changed this test function to test both with views and columns, which catches the issue with current gtests. So it seems like it is probably a worthwhile change.

```
[ RUN      ] SplitStringTest.StringWithInvalids
/home/abellina/work/cudf/cpp/tests/utilities/column_utilities.cu:41: Failure
      Expected: lhs.size()
      Which is: 3
To be equal to: rhs.size()
      Which is: 1
/home/abellina/work/cudf/cpp/tests/utilities/column_utilities.cu:42: Failure
      Expected: lhs.null_count()
      Which is: 1
To be equal to: rhs.null_count()
      Which is: 0
/home/abellina/work/cudf/cpp/tests/utilities/column_utilities.cu:44: Failure
      Expected: lhs.nullable()
      Which is: true
To be equal to: rhs.nullable()
      Which is: false
/home/abellina/work/cudf/cpp/tests/utilities/column_utilities.cu:41: Failure
      Expected: lhs.size()
      Which is: 3
To be equal to: rhs.size()
      Which is: 1
/home/abellina/work/cudf/cpp/tests/utilities/column_utilities.cu:42: Failure
      Expected: lhs.null_count()
      Which is: 1
To be equal to: rhs.null_count()
      Which is: 0
/home/abellina/work/cudf/cpp/tests/utilities/column_utilities.cu:44: Failure
      Expected: lhs.nullable()
      Which is: true
To be equal to: rhs.nullable()
      Which is: false
unknown file: Failure
C++ exception with description "cuDF failure at: /home/abellina/work/cudf/cpp/src/strings/copying/copying.cu:53: Invalid start parameter value." thrown in the test body.

```

Reply:
@jrhemstad happy to make this different.

Reply:
I think he wants to check whether column derived from column view doesn't change or break.

Reply:
This feels like something that should be in a specific unit test, not in the utility function that's used by ALL unit tests.

Reply:
What @ChuckHastings said.

Reply:
Ok, that sounds good. @nvdbaranec had a similar reaction. I'll continue to use the function with column_view but call it using a column (so it casts to a view). This, I just verified, also shows the same test failure.

Reply:
https://github.com/rapidsai/cudf/pull/3886/commits/c1cb44b960483a054a2544d3a7213dad08dd2967 does this, and removes the change to the utility function.